velox/expression/CastExpr.cpp

```diff
@@ -46,6 +47,17 @@ void applyCastKernel(
     vector_size_t row,
     const SimpleVector<typename TypeTraits<FromKind>::NativeType>* input,
     FlatVector<typename TypeTraits<ToKind>::NativeType>* result) {
+  if (input->type()->isDecimal()) {
```

Comment:
Do we need a compile time check on input kind?

Reply:
You mean why is it a runtime check? Decimal Types are logical types and there are no TypeKinds mapped to them. So we can't make it a compile time check. The mean runtime is 10183928.214538805 ns vs 101502160.15259695 ns. There is a difference of 0.3ms. CMIW.

I do see some cast operations in the benchmarking that could hit this check and result in slowdown:

```
BENCHMARK(multiplyLiteralSmall) {
  benchmark->runSmall("multiply(a, cast(2 as double))");
}

BENCHMARK(multiplyLiteralMedium) {
  benchmark->runMedium("multiply(a, cast(2 as double))");
}

BENCHMARK(multiplyLiteralLarge) {
  benchmark->runLarge("multiply(a, cast(2 as double))");
}
```

Reply:
I will try to isolate applyCastKernel for Decimal case into a separate method may be, so that the runtime check is not made for every row.